context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx

```diff
@@ -0,0 +1,108 @@
+import React, { useReducer, useRef } from 'react';
+
+import ShareIcon from '@material-ui/icons/Share';
+import Popper from '@material-ui/core/Popper';
+import Paper from '@material-ui/core/Paper';
+import MenuList from '@material-ui/core/MenuList';
+import MenuItem from '@material-ui/core/MenuItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ClickAwayListener from '@material-ui/core/ClickAwayListener';
+import Link from '@material-ui/core/Link';
+import { encodeConfInUrl } from 'vitessce';
+
+import { SecondaryBackgroundTooltip } from 'js/shared-styles/tooltips';
+import useVisualizationStore from 'js/stores/useVisualizationStore';
+import { WhiteBackgroundIconButton } from 'js/shared-styles/buttons';
+
+import { StyledLinkIcon, StyledTypography, StyledEmailIcon } from './style';
+import 'vitessce/dist/es/production/static/css/index.css';
+
+const DEFAULT_LONG_URL_WARNING =
+  'Warning: this is a long URL which may be incompatible or load slowly with some browsers.';
+
+const DEFAULT_EMAIL_MESSAGE = 'Here is an interesting dataset I found in the HuBMAP Data Portal:';
+
+const visualizationStoreSelector = (state) => ({
+  vitessceConfig: state.vitessceConfig,
+  setOnCopyUrlWarning: state.setOnCopyUrlWarning,
+  setOnCopyUrlSnackbarOpen: state.setOnCopyUrlSnackbarOpen,
+});
+function VisualizationShareButton() {
+  const [open, toggle] = useReducer((v) => !v, false);
+  const anchorRef = useRef(null);
+  const { vitessceConfig, setOnCopyUrlWarning, setOnCopyUrlSnackbarOpen } = useVisualizationStore(
+    visualizationStoreSelector,
+  );
+
+  const copyToClipBoard = (conf) => {
+    const dummy = document.createElement('input');
+    document.body.appendChild(dummy);
+    const url = `${window.location.href.split('#')[0]}#${encodeConfInUrl({
+      conf,
+      onOverMaximumUrlLength: () => {
+        setOnCopyUrlWarning(DEFAULT_LONG_URL_WARNING);
+      },
+    })}`;
```

Comment:
```suggestion
    const encodedConf = encodeConfInUrl({
      conf,
      onOverMaximumUrlLength: () => {
        setOnCopyUrlWarning(DEFAULT_LONG_URL_WARNING);
      },
    });
    const url = `${window.location.href.split('#')[0]}#${encodedConf}`;
```

Too much nesting gets hard to read.

Reply:
Actually, pull this out as a utility function, since it's copy and pasted below.